### PR TITLE
add support for fmi2GetDirectionalDerivative

### PR DESCRIPTION
**Pull request by arun3688**
### Purpose

This PR implements the `partial derivatives of fmu signals` through  `fmi2GetDirectionalDerivative`.  This PR is Work in Progress

### TODO

- [x] Implement the `fmi2GetDirectionalDerivative` table according to FMI specification
- [x] Add tests and verify the results
- [x] Add python support to API
- [x] Add Documentation

### CODE REVIEW

File: src/OMSimulatorLib/ComponentFMUCS.cpp [hunk 1]
```diff
@@ -572,7 +572,7 @@ oms_status_enu_t oms::ComponentFMUCS::instantiate()
       for (const auto &res : it.allresources)
       {
         if (res.second.linkResources) // set values only if resources are linked in ssd
-          setResourcesHelper1(res.second);
+            setResourcesHelper1(res.second);
```
Comment (lochel) [inline src/OMSimulatorLib/ComponentFMUCS.cpp:575]: Changing the indent here is just a mistake, isn't it?

File: src/OMSimulatorLib/ComponentFMUCS.cpp [hunk 2]
```diff
@@ -1146,6 +1146,110 @@ oms_status_enu_t oms::ComponentFMUCS::getReal(const ComRef& cref, double& value)
...
+    }
+  }
+
+  if (!fmu || j < 0)
+    return logError_UnknownSignal(getFullCref() + cref);
+
+  if (oms_modelState_instantiated == getModel().getModelState())
+  {
+    if (getFMUInfo()->getGenerationTool().substr(0, 12) == "OpenModelica")
+      logWarning("It is not possible to get partial derivatives of OpenModelica generated fmus during initialization mode, getting directional derivative after intialization is possible");
```
Comment (lochel) [inline src/OMSimulatorLib/ComponentFMUCS.cpp:1174]: I expected this to become a warning within the fmu and not OMSimulator. But doesn't matter I guess, since we are going to implement support for it ASAP anyways :)

File: testsuite/api/Makefile [hunk 3]
```diff
@@ -14,6 +14,8 @@ deleteReferencesInSSD1.lua \
 deleteReferencesInSSD2.lua \
 deleteResourcesInSSP1.lua \
 deleteResourcesInSSP2.lua \
+getDirectionalDerivative_CS.lua \
+getDirectionalDerivative_ME.lua \
```
Comment (lochel) [inline testsuite/api/Makefile:18]: The test is not part of this pull request. Maybe you forgot to add the file?

**Review by lochel — APPROVED** [review]

File: testsuite/api/Makefile (hunk 3 above)
Comment (arun3688) [reply testsuite/api/Makefile:18]: Yes you are correct `getDirectionalDerivative_ME` is missing , i will add it

File: src/OMSimulatorLib/ComponentFMUCS.cpp (hunk 1 above)
Comment (arun3688) [reply src/OMSimulatorLib/ComponentFMUCS.cpp:575]: Yes it was an mistake

File: testsuite/api/Makefile (hunk 3 above)
Comment (arun3688) [reply testsuite/api/Makefile:18]: this is fixed with this commit https://github.com/OpenModelica/OMSimulator/pull/1109/commits/12630fe699e544e6f1c3c1969cd661c5ef79d0a2
